File: s06_RecyclerView/LandmarkBook/app/src/main/java/com/omeryildizce/landmarkbook/MainActivity.java
package com.omeryildizce.landmarkbook;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    Landmark pisa;
    Landmark eiffel;
    Landmark colloseum;
    Landmark londonbridge;
    ArrayList<Landmark> landmarks;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Data
        pisa = new Landmark("Pisa", "Italy", R.drawable.pisa);
        eiffel = new Landmark("Eiffel", "France", R.drawable.eiffel);
        colloseum = new Landmark("Colloseum", "Italy", R.drawable.colloseum);
        londonbridge = new Landmark("London Bridge", "England", R.drawable.londonbridge);

        landmarks = new ArrayList<Landmark>();
        landmarks.add(pisa);
        landmarks.add(eiffel);
        landmarks.add(colloseum);
        landmarks.add(londonbridge);
    }
}

File: s19_LandmarkBook/LandmarkBook/app/src/main/java/com/omeryildizce/landmarkbook/MainActivity.kt
package com.omeryildizce.landmarkbook

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.AdapterView
import android.widget.ArrayAdapter
import androidx.recyclerview.widget.LinearLayoutManager
import com.omeryildizce.landmarkbook.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    private lateinit var landmarkList: ArrayList<Landmark>
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)
        // Load Data
        loadLandmarkData()
        // listView()

        // recycler view
        recyclerView()
    }

    private fun recyclerView() {
        binding.recyclerView.layoutManager = LinearLayoutManager(this@MainActivity)
        val landmarkAdapter = LandmarkAdapter(landmarkList)
        binding.recyclerView.adapter = landmarkAdapter
    }

    /*
    private fun listView() {
        // Adapter : Layout : Data
        val adapter =
            ArrayAdapter(
                this@MainActivity,
                android.R.layout.simple_list_item_1,
                landmarkList.map { landmark ->
                    // Mapping
                    landmark.name
                })
        binding.listView.adapter = adapter
        binding.listView.onItemClickListener =
            AdapterView.OnItemClickListener { parent, view, position, id ->
                val intent = Intent(this@MainActivity, DetailsActivity::class.java)
                intent.putExtra("landmark", landmarkList.get(position))
                startActivity(intent)

            }
    }
    */

    private fun loadLandmarkData() {
        val pisa = Landmark("Pisa", "Italy", R.drawable.pisa)
        val colloseum = Landmark("Colloseum", "Italy", R.drawable.colloseum)
        val eiffel = Landmark("Eiffel", "France", R.drawable.eiffel)
        val londonbridge = Landmark("London Bridge", "England", R.drawable.londonbridge)
        
        landmarkList = ArrayList<Landmark>()
        landmarkList.add(pisa)
        landmarkList.add(eiffel)
        landmarkList.add(colloseum)
        landmarkList.add(londonbridge)
    }
}